src/components/Coldcard/ColdcardExtendedPublicKeyImporter.jsx

```diff
@@ -0,0 +1,111 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { COLDCARD } from "unchained-wallets";
+import { FormGroup, FormHelperText } from "@material-ui/core";
+import { MAINNET, P2SH } from "unchained-bitcoin";
+import { ColdcardJSONReader } from ".";
+import IndirectExtendedPublicKeyImporter from "../Wallet/IndirectExtendedPublicKeyImporter";
+
+class ColdcardExtendedPublicKeyImporter extends React.Component {
+  constructor(props) {
+    super(props);
+    const coldcardBIP32Path = this.getColdcardBip32Path();
+    this.state = {
+      COLDCARD_MULTISIG_BIP32_PATH: coldcardBIP32Path,
+    };
+  }
+
+  componentDidMount = () => {
+    const { extendedPublicKeyImporter, validateAndSetBIP32Path } = this.props;
+    const { COLDCARD_MULTISIG_BIP32_PATH } = this.state;
+    if (extendedPublicKeyImporter.method === COLDCARD) {
+      validateAndSetBIP32Path(
+        COLDCARD_MULTISIG_BIP32_PATH,
+        () => {},
+        () => {},
+        {}
+      );
+    }
+  };
+
+  componentDidUpdate(prevProps) {
+    const { validateAndSetBIP32Path, network, addressType } = this.props;
+    const coldcardBIP32Path = this.getColdcardBip32Path();
+
+    // Any updates to the network/addressType we should set the BIP32Path
+    if (
+      prevProps.network !== network ||
+      prevProps.addressType !== addressType
+    ) {
+      // eslint-disable-next-line react/no-did-update-set-state
+      this.setState({ COLDCARD_MULTISIG_BIP32_PATH: coldcardBIP32Path });
```

Comment:
Couple points: 
1. Just a reminder that this setState won't necessarily get fired before the `validateAndSetBIP32Path` call below. Not sure it will make a difference in this particular case, but something to be aware of. If order is important, I'd recommend passing it in as a callback as the second arg to `setState`. In fact, don't want we want to validate and set before we set it locally? If there is a problem with it couldn't that cause some weird behavior?
2. Again, shouldn't be a blocker, but as an opportunity to point this out, this component could be a good candidate for using hooks. The `useEffect` hook allows you to be explicit about what the callback would react to. The reason I point this out is that the linter is calling this out because `setState` is going to cause another update which calls this lifecycle method again. If the `if` statement guard isn't careful enough, this results in an infinite loop. `useEffect` lets you be explicit to hopefully avoid that.

Reply:
Agree with these comments. 

There's a bit of an issue with the fact that we can't have Coldcard (in the P2SH) conform to our multisig standard P2SH path of `m/45'/{0-1}'/0'` as the root/base path. So what actually ends up happening here when this loads is the interaction class is initially *not* supported because the bip32 validation fails for the coldcard. (https://github.com/unchained-capital/unchained-wallets/blob/master/src/coldcard.js#L153) -- This gets fixed once this setState does actually fire with the `m/45'/{0-1}/0` path. and because it's current UNSUPPORTED, we check with unchained-wallets again.